73_ArithmeticBench/app_resources/shaderCommon.hlsl

```diff
@@ -0,0 +1,85 @@
+#include "common.hlsl"
+
+#include "nbl/builtin/hlsl/glsl_compat/core.hlsl"
+#include "nbl/builtin/hlsl/subgroup/basic.hlsl"
+#include "nbl/builtin/hlsl/subgroup/arithmetic_portability.hlsl"
+#include "nbl/builtin/hlsl/subgroup2/arithmetic_portability.hlsl"
+
+#include "nbl/builtin/hlsl/jit/device_capabilities.hlsl"
+
+// https://github.com/microsoft/DirectXShaderCompiler/issues/6144
+uint32_t3 nbl::hlsl::glsl::gl_WorkGroupSize() {return uint32_t3(WORKGROUP_SIZE,1,1);}
+
+// unfortunately DXC chokes on descriptors as static members
+// https://github.com/microsoft/DirectXShaderCompiler/issues/5940
+[[vk::binding(0, 0)]] StructuredBuffer<uint32_t> inputValue;
+[[vk::binding(1, 0)]] RWByteAddressBuffer output[8];
+
+// because subgroups don't match `gl_LocalInvocationIndex` snake curve addressing, we also can't load inputs that way
+uint32_t globalIndex();
```

Comment:
rename to `globalFirstItemIndex` to  mean that its the first index to access by the invocation, and add a comment that to get next item one does not scroll by 1 but by SubgroupSize

Reply:
bump